### DISCUSSION TITLE
list of match pairs to avoid exhastive matching of all possible pairs

### DISCUSSION THREAD

**Discussion by avispi** [openMVG/openMVG#1998] [Q&A] [answered]
Hi.
from the prints, I understand such file can be given as input to save processing time.
How should it be done? what is the format?
Thank you.

**Comment by pmoulon** [comment, marked as answer]
You can provide a pair file with the id of the view (see the id in the sfm_data.json files).
Note: Images are sorted by alphabetical order

See here how to provide you pair list:
https://github.com/openMVG/openMVG/blob/develop/src/software/SfM/main_ComputeMatches.cpp#L62


Note that OpenMVG has different mechanisms for the matching pair:
- exhaustive
- video
- preemptive
- VLAD
- GPS
See [here](https://www.linkedin.com/posts/pierre-moulon_openmvg-v20-rainbow-trout-is-out-whats-activity-6861444983413714944-16Hm) for the visual illustration 

The pair file could be as follow (number of pair and then index of the view -> sfm_data.json):
```
3
1 2
2 3
3 4
```
